spec/ddtrace/configuration_spec.rb

```diff
@@ -9,6 +10,140 @@
 
     describe '#configure' do
```

Comment:
Most of these tests are "integration-like" in the sense that they apply some high level settings and consequently run through other components like `Configuration::Settings` and `Configuration::Components` and assert on the stateful outcome of those classes.

Might want to extract this kind of behavioral test elsewhere, but put it in here for now, to ensure basic usage works as intended.